src/Factory/DoctrineQueueFactory.php

```diff
@@ -28,16 +28,17 @@ public function __invoke(ContainerInterface $container, $requestedName, array $o
         $connection       = $container->get($queueOptions->getConnection());
         $jobPluginManager = $container->get(JobPluginManager::class);
 
-        $queue = new DoctrineQueue($connection, $queueOptions, $requestedName, $jobPluginManager);
-
-        return $queue;
+        return new DoctrineQueue($connection, $queueOptions, $requestedName, $jobPluginManager);
     }
 
     /**
      * {@inheritDoc}
      */
-    public function createService(ServiceLocatorInterface $serviceLocator, $name = '', $requestedName = '')
-    {
+    public function createService(
```

Comment:
Once you raise your dependencies to a level that is PHP 7.2 compatible I think you will find that these no longer this backwards compatibility layer. I would advise to migrate to the new style, and remove the old `createService` functions.